File: SDKs/amazonaws.com/codestar-notifications/2019-10-15/java/lib/src/main/java/openapisdk/models/shared/TargetSummary.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * TargetSummary
 * Information about the targets specified for a notification rule.
**/
public class TargetSummary {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("TargetAddress")
    public String targetAddress;
    public TargetSummary withTargetAddress(String targetAddress) {
        this.targetAddress = targetAddress;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("TargetStatus")
    public TargetStatusEnum targetStatus;
    public TargetSummary withTargetStatus(TargetStatusEnum targetStatus) {
        this.targetStatus = targetStatus;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("TargetType")
    public String targetType;
    public TargetSummary withTargetType(String targetType) {
        this.targetType = targetType;
        return this;
    }
}

File: SDKs/amazonaws.com/codestar-notifications/2019-10-15/python/src/sdk/models/shared/targetsummary.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import targetstatus_enum as shared_targetstatus_enum


@dataclass_json
@dataclasses.dataclass
class TargetSummary:
    r"""TargetSummary
    Information about the targets specified for a notification rule.
    """
    
    target_address: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('TargetAddress') }})
    target_status: Optional[shared_targetstatus_enum.TargetStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('TargetStatus') }})
    target_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('TargetType') }})
    
